internal/openshiftcontroller/openshiftcontroller.go

```diff
@@ -312,124 +290,33 @@ func (oc *OpenShiftController) CheckNewUser(ns string) (err error) {
 	return
 }
 
-//ProcessBuilds processes builds when not using `watch` (i.e. using polling)
-//and updates user's structure
-func (oc *OpenShiftController) processBuilds(namespaces []string) (err error) {
-	for _, n := range namespaces {
-		err := oc.CheckNewUser(n)
-		if err != nil {
-			return err
-		}
-
-		bl, err := oc.o.GetBuilds(n)
-		if err != nil {
-			log.Error("Could not load builds: ", err)
-			continue
-		}
-
-		lastActive := oc.Users[n].ActiveBuild
-		lastDone := oc.Users[n].DoneBuild
-		for i := range bl.Items {
-			if IsActive(&bl.Items[i]) {
-				lastActive, err = GetLastBuild(lastActive, &bl.Items[i])
-			} else {
-				lastDone, err = GetLastBuild(lastDone, &bl.Items[i])
-			}
-			if err != nil {
-				log.Error(err)
-			}
-		}
-		oc.lock.Lock()
-		*oc.Users[n].ActiveBuild = *lastActive
-		*oc.Users[n].DoneBuild = *lastDone
-		oc.lock.Unlock()
-	}
-
-	return
-}
-
 //Run implements main loop of the application
-func (oc *OpenShiftController) Run(groupNumber int) {
-	//Use watch instead of polling
-	if oc.watch {
-		//FIXME - this looks ugly
-		go func() {
-			for {
-				//For each user we know about, check if there is any action needed
-				for _, u := range oc.Users {
-					enabled, err := oc.features.IsIdlerEnabled(u.ID)
-					if err != nil {
-						log.Error("Error checking for idler feature", err)
-						continue
-					}
-					if !enabled {
-						log.Debugf("Skipping check for %s.", u.Name)
-						continue
-					}
-					err = oc.CheckIdle(u)
-					if err != nil {
-						log.Errorf("Could not check idling for %s: %s", u.Name, err)
-					}
+func (oc *OpenShiftController) Run() {
+	go oc.o.WatchDeploymentConfigs("", "-jenkins", oc.HandleDeploymentConfig)
```

Comment:
+1 for starting the monitoring/watching first and then the for loop. Easier to understand.